lib/engine/game/g_1848/step/buy_train.rb

```diff
@@ -17,14 +31,23 @@ def can_entity_buy_train?(entity)
           def buyable_trains(entity)
             # Cannot buy 2E if one is already owned
             owns_2e = entity.trains.any? { |t| t.name == '2E' }
-            return super unless owns_2e
-
-            super.reject { |t| t.name == '2E' }
+            trains_to_buy = super
+            trains_to_buy = trains_to_buy.select(&:from_depot?) unless @game.phase.status.include?('can_buy_trains')
+            trains_to_buy = trains_to_buy.reject { |t| t.name == '2E' } if owns_2e
+            trains_to_buy
           end
 
           def room?(entity)
             entity.trains.count { |t| t.name != '2E' } < @game.train_limit(entity)
```

Comment:
Are you handling '2E' in crowded corp logic within the game class?

Reply:
yeah, right here: https://github.com/tobymao/18xx/blob/963040057e6a8b7d3ca74ac991def3a95c53343f/lib/engine/game/g_1848/game.rb#L638